test_scripts/AppIconsFolder/ATF_P_AppIconsFolder_exists_with_ReadWritePermissions.lua

```diff
@@ -0,0 +1,144 @@
+end
+
+local function pathToAppFolderFunction(appID)
+  return commonPreconditions:GetPathToSDL() .. "storage/" .. appID .. "_" .. config.deviceMAC .. "/"
+end
+
+local function checkFolderCreated(FolderPath)
+  local returnValue = true
+  local command = assert( io.popen(  "[ -d " .. FolderPath .. " ] && echo \"Exist\" || echo \"NotExist\"" , 'r'))
+  os.execute("sleep 1")
+  local commandResult = command:read( '*l' )
+    if commandResult == "NotExist" then
+      returnValue = false
+    elseif commandResult == "Exist" then
+     returnValue = true
+    end
+    return returnValue
+end 
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_connectMobile()
+  self:connectMobile()
+end
+
+function Test:Precondition_RegisterAppWithIcon()
+  local pathToAppFolder = pathToAppFolderFunction(RAIParameters.appID)
+  self.mobileSession= mobile_session.MobileSession(self, self.mobileConnection)
+  self.mobileSession.version = 4
+  self.mobileSession:StartService(7)
+  :Do(function()
+    registerApplication(self)
+    EXPECT_NOTIFICATION("OnHMIStatus", { systemContext = "MAIN", hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE"})
+     :Do(function()
+     local cidPutFile = self.mobileSession:SendRPC("PutFile",
+       {
+         syncFileName = "icon.png",
+         fileType = "GRAPHIC_PNG",
+         persistentFile = false,
+         systemFile = false
+       }, "files/icon.png")
+     EXPECT_RESPONSE(cidPutFile, { success = true, resultCode = "SUCCESS" })
+     :Do(function()
+       local cidSetAppIcon = self.mobileSession:SendRPC("SetAppIcon",{ syncFileName = "icon.png" })
+        EXPECT_HMICALL("UI.SetAppIcon",
+         {
+           syncFileName =
+            {
+              imageType = "DYNAMIC",
+              value = pathToAppFolder .. "icon.png"
+            }
+         })
+         :Do(function(_,data)
+           self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+          end)
+         EXPECT_RESPONSE(cidSetAppIcon, { resultCode = "SUCCESS", success = true })
+         end)
+       end)
+    end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test.Check_SDL_finds_icons_saved_in_AppIconsFolder()
+  local status = true
+  local dirExistResult = checkFolderCreated(commonPreconditions:GetPathToSDL() .. "storage")
+  if dirExistResult == true then
+    local applicationFileToCheck = commonPreconditions:GetPathToSDL() .. "storage/" .. RAIParameters.appID
+    local applicationFileExistsResult = commonSteps:file_exists(applicationFileToCheck)
+    if applicationFileExistsResult == false then
+      commonFunctions:userPrint(31, RAIParameters.appID .. " icon is not written to folder")
+      status = false
+    end
+    else 
+      commonFunctions:userPrint(31, "storage folder does not exist in SDL bin folder" )
+      status = false
+    end
+    return status
```

Comment:
@aarotar, In order to have test step failed we have to use self:FailTestCase() function